File: nvxapp.server.service/ClientServer-Service/Account/Models/UserCompanyModel.cs
using nvxapp.server.data.Entities.Public;
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static nvxapp.server.data.Entities.AspNetUsersDataUtil;

namespace nvxapp.server.service.ClientServer_Service.Account.Models
{
    public class UserCompanyModel
    {
        public string? IdAspNetUsers { get; set; }

        public string? Descrizione { get; set; }
    }
    public class UserCompanyListInModel
    {
    }
    public class UserCompanyListOutModel : ModelResult
    {
        public List<UserCompanyModel> UserCompanyList { get; set; }

        public UserCompanyListOutModel()
        {
            UserCompanyList = new List<UserCompanyModel>();
        }
    }



    public class UserCompanyEditModel
    {
        public string IdAspNetUsers { get; set; } = string.Empty;
        public string Descrizione { get; set; } = string.Empty;
    }
    public class UserCompanyGetInModel
    {
        public string Id { get; set; } = string.Empty ;
    }
    public class UserCompanyGetOutModel : ModelResult
    {
        public UserCompanyEditModel UserCompanyEdit { get; set; } = new UserCompanyEditModel();
    }
    public class UserCompanyPutInModel : ModelResult
    {
        public UserCompanyEditModel UserCompanyEdit { get; set; } = new UserCompanyEditModel();
    }
    public class UserCompanyPutOutModel : ModelResult
    {
        public UserCompanyEditModel UserCompanyEdit { get; set; } = new UserCompanyEditModel();
    }

}

File: nvxapp.client/src/nvx/ClientServer-Service/Infrastructure/Account/Models/user-company-model.ts
import { Dip_AnagraficaModel } from "../../../GestionePresenze/Dip_Anagrafica/Models/dip-anagrafica-model";
import { Dip_ProfiloOrarioModel } from "../../../GestionePresenze/Dip_ProfiloOrario/Models/dip-profilo-orario-model";
import { Dip_RapportoLavoroModel } from "../../../GestionePresenze/Dip_RapportoLavoro/Models/dip-rapporto-lavoro-model";
import { ModelResult } from "../../../ModelsBase/model-result";
import { RolesModel } from "./user-roles-model";


export class UserCompanyListModel {
  constructor(
    public descrizione: string | null = "",
    public idAspNetUsers: string = "",
    public idUserCompany: number = 0,
    public mainUser: boolean = false,
    
    public roles: string[] = []
  ) { }
}
export class UserCompanyListInModel {

  constructor(
    public filteredRoles: string[] = []
  ) { }

}
export class UserCompanyListOutModel extends ModelResult {

  public userCompanyList: UserCompanyListModel[] ;

}


 
  


export class UserCompanyEditModel {
  constructor(
    public descrizione: string | null = "",
    public idUserCompany: number = 0,
    public idAspNetUsers: string = "",
    public mainUser: boolean = false,

    public mail: string | null = null,
    public pw: string | null = null,
    
    public roles: string[] = []

    
  ) { }
}

export class UserCompanyGetInModel {
  public id: number;
}
export class UserCompanyGetOutModel extends ModelResult {

  public userCompanyEdit: UserCompanyEditModel;

}
export class UserCompanyPutInModel  {

  public userCompanyEdit: UserCompanyEditModel;

}
export class UserCompanyPutOutModel extends ModelResult {

  public userCompanyEdit: UserCompanyEditModel;

}
